src/registration.rs

```diff
@@ -358,17 +362,17 @@ impl Registered {
              redirect_uri={}",
             self.base, self.client_id, self.client_secret, code.as_ref(), self.redirect
         );
-        debug!(url = url; "completing registration");
+        debug!(url, "completing registration");
         let response = self.client.post(&url).send().await?;
         debug!(
-            status = log_serde!(response Status), url = url,
-            headers = log_serde!(response Headers);
+            url,
+            response = as_value!(response, Response),
             "received API response"
         );
         let token: AccessToken = read_response(response).await?;
-        debug!(url = url, body = as_serde!(token); "parsed response body");
+        debug!(url, body = ?token, "parsed response body");
```

Comment:
Since this is a singleton struct, why not this?
```suggestion
        debug!(url, token = %token.access_token, "received access token");
```

Although maybe it would be better not to log that at all.

Reply:
It's definitely better to not log this at all. I think I later ripped it out or made it trace only in the spans code.